allauth/socialaccount/providers/battlenet/provider.py

```diff
@@ -5,17 +5,13 @@
 
 
 class BattleNetSocialAccountAdapter(DefaultSocialAccountAdapter):
```

Comment:
We can drop the adapter if we do one of two things:

* Either we change the default username regex to accept `#` as a valid character
* Or we build a mechanism for socialaccount providers to somehow accept a custom username regex.

The former would be easy, but affect all other providers. The latter would be fairly hard, and I'm not up to the task.

Regardless of the outcome, this PR should be merged IMO as it's more correct than what's on master right now and prevents a crash.